src/NHibernate.Validator/Mappings/MixedClassMapping.cs

```diff
@@ -22,37 +24,59 @@ protected virtual void InitializeClassAttributes(IClassMapping baseMap, IClassMa
 
 		protected void MixMembersWith(HashSet<MemberInfo> lmembers, IClassMapping mapping)
 		{
-			foreach (MemberInfo info in mapping.GetMembers())
+			foreach (var info in mapping.GetMembers())
 			{
 				lmembers.Add(info);
-				IEnumerable<Attribute> mas = mapping.GetMemberAttributes(info);
-				if (mas != null)
+				var mas = mapping.GetMemberAttributes(info);
+				if (mas == null)
+					continue;
+
+				if (!membersAttributesDictionary.TryGetValue(info, out var attrs))
 				{
-					List<Attribute> attrs;
-					if (!membersAttributesDictionary.TryGetValue(info, out attrs))
-					{
-						membersAttributesDictionary[info] = new List<Attribute>(mas);
-					}
-					else
-					{
-						CombineAttribute(mas, attrs);
-						membersAttributesDictionary[info] = attrs;
-					}
+					membersAttributesDictionary[info] = new List<Attribute>(mas);
+				}
+				else
+				{
+					CombineAttribute(mas, attrs);
+					membersAttributesDictionary[info] = attrs;
 				}
 			}
 		}
 
 		protected static void CombineAttribute(IEnumerable<Attribute> origin, List<Attribute> dest)
 		{
-			foreach (Attribute ma in origin)
+			foreach (var ma in origin)
 			{
-				Attribute found = dest.Find(attribute => ma.TypeId.Equals(attribute.TypeId));
+				var founded = dest.FindAll(attribute => ma.TypeId.Equals(attribute.TypeId));
 
-				if (found != null && !AttributeUtils.AttributeAllowsMultiple(ma))
-					dest.Remove(found);
+				if (founded.Count > 0)
+				{
+					if (!AttributeUtils.AttributeAllowsMultiple(ma))
+					{
+						dest.Remove(founded[0]);

```

Comment:
Shouldn't it remove all instead of just the first? Maybe in that case there is only one anyway but well, better have code reflecting the intent as most as possible.